app/src/main/java/com/babylon/wallet/android/domain/model/factors/FactorSourceCard.kt

```diff
@@ -0,0 +1,27 @@
+package com.babylon.wallet.android.domain.model.factors
+
+import com.radixdlt.sargon.Account
+import com.radixdlt.sargon.FactorSourceId
+import com.radixdlt.sargon.FactorSourceKind
+import com.radixdlt.sargon.Persona
+import kotlinx.collections.immutable.PersistentList
+
+data class FactorSourceCard(
+    val kind: FactorSourceKind,
+    val header: Header,
+    val messages: PersistentList<StatusMessage>,
+    val accounts: PersistentList<Account>,
+    val personas: PersistentList<Persona>
+) {
+
+    sealed interface Header {
+
+        data object New : Header
+
+        data class Instance(
+            val id: FactorSourceId.Hash,
+            val name: String,
+            val lastUsedOn: String?
```

Comment:
Is this supposed to be a string or a Sargon timestamp?

Reply:
This is the formatted time based on a Sargon Timestamp. It's a matter of deciding if we want to format it in the ViewModel or directly in the UI, now  I'm formatting it in the ViewModel and let the UI display it as it is.